packages/flutter_tools/lib/src/commands/screenshot.dart

```diff
@@ -183,6 +191,38 @@ class ScreenshotCommand extends FlutterCommand {
     return true;
   }
 
+  /// Additional error checks below for valid file name
+  /// and directory because ios/xcode screenshot binary will
+  /// return a 0 exit code even if outputFile is not valid
+  static void validateOutputFile(File outputFile, FileSystem fs) {
+
+    const Platform platform = LocalPlatform();
+
+    // Conditional to check for trailing path separator
+    if (outputFile.path.endsWith(platform.pathSeparator)) {
+      throwToolExit(
+        'The provided path cannot end with a path separator\n'
+        'Path provided: "${outputFile.path}"'
```

Comment:
I would rephrase this to indicate that we expect a file path but got a directory path.